experimentation/build.gradle

```diff
@@ -45,6 +46,10 @@ dependencies {
     //WordPress utils
     implementation "org.wordpress:utils:2.3.0"
 
+    //Dagger
+    implementation "com.google.dagger:dagger:2.41"
+    kapt "com.google.dagger:dagger-compiler:2.41"
```

Comment:
👋 I'm concerned about using `Dagger` in such small library. I understand the purpose of using `Lazy` but I don't think we need to use such powerful tool in this case.

Also it feels dangerous to me: if I'm not mistaken, we loose compile-time validation when injecting `Experiment`s into DI graph. E.g. we can add an `Experiment` in scope of `Fragment`, while `ExPlat` has been created in scope of `Application`. In this case I believe we'll have only `Experiment`s that were created during time of creating `Application` (`Lazy#get` [will refer to the initially injected set](https://dagger.dev/api/2.13/dagger/Lazy.html)).

While it's fixable (inject all experiments in the same scope as `ExPlat`), I think it makes API more difficult and tricky to work with.

---

I don't believe that `Experiment` needs `ExPlat` dependency. WDYT about:

```kotlin
// If I understand correctly, this class is meant to be inherited from
open class Experiment(
    val name: String,
)
```

and then in `ExPlat.kt`

```kotlin 
fun getVariation(experiment: Experiment, shouldRefreshIfStale: Boolean): Variation
```

WDYT?

Reply:
FWIW I think these are valid concerns. The original implementation using `Lazy` wasn't ideal – even considering it wasn't being used in a library.

If I'm not mistaken, the idea of having `ExPlat` as a dependency of `Experiment` was to make it clearer to understand where each experiment was being used, as a class would explicitly declare its dependency on an individual experiment instead of `ExPlat`. We can probably achieve a similar result by refactoring `ExPlat` with a better separation of responsibilities, but the way I see it, this is more of a nice-to-have than a must-have, so I agree with the proposal of simply removing the `ExPlat` dependency from `Experiment`.

One thing I want to add is that if we decide to go down this route, I would also consider making `Experiment` an interface instead of an open class.